upup/models/cloudup/resources/addons/networking.projectcalico.org/k8s-1.12.yaml.template

```diff
@@ -240,9 +240,12 @@ data:
   # essential.
   typha_service_name: "none"
 
-  # TODO: Do we want to configure this?
   # Configure the MTU to use
-  veth_mtu: "1440"
+  {{- if .Networking.Calico.MTU }}
+  veth_mtu: "{{ .Networking.Calico.MTU }}"
+  {{- else }}
+  veth_mtu: "{{- if eq .CloudProvider "openstack" -}}1430{{- else -}}1440{{- end -}}"
```

Comment:
Part of me wonders if this can all be avoided by doing the logic upstream of setting MTU if not set to either 1440 or 1430 depending on provider. That said I can't think of a straight forward spot to put that right now so this is probably okay.

Reply:
I have tested this line and it worked well